api/tasks/runner.go

```diff
@@ -439,3 +446,15 @@ func (t *task) envVars(home string, pwd string, gitSSHCommand *string) []string
 
 	return env
 }
+
+// checkTmpDir checks to see if the temporary directory exists
+// and if it does not attempts to create it
+func checkTmpDir(path string) error {
+	var err error = nil
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			return os.MkdirAll(path, 1777)
```

Comment:
Is it safe? If we want to store keys in the temporary location they may be found with simple `find`, if we make this directory world-readable.

Reply:
well... this is an issue actually because if you are writing to tmp it must be 1777.
However it would be simple enough to see if the first part of the path matches the os.TmpDir or "/tmp" (because os.TmpDir could be $TMPDIR) and then change the permissions accordingly. I would suggest either 0660 or 0640 for non tmp folders, what do you think?